src/sage/schemes/hyperelliptic_curves/jacobian_endomorphisms.py

```diff
@@ -0,0 +1,465 @@
+
+    - ``C`` -- the hyperelliptic curve.
+
+    - ``bad_primes`` -- the list of odd primes of bad reduction.
+
+    - ``B`` -- the bound which appears in the statement of the algorithm
+    from Lombardo's paper.
+
+    OUTPUT:
+
+        (bool1, bool2): Pair of booleans. `bool1` indicates if the
+        absolute endomorphism algebra is a field; `bool2` indicates if the
+        absolute endomorphism algebra is the field of rational numbers.
+
+    EXAMPLES::
+
+    This is LMFDB curve 940693.a.960693.1
+
+        sage: from sage.schemes.hyperelliptic_curves.jacobian_endomorphisms import get_is_geom_irred
+        sage: R.<x> = QQ[]
+        sage: f = 4*x^6 - 12*x^5 + 20*x^3 - 8*x^2 - 4*x + 1
+        sage: C = HyperellipticCurve(f)
+        sage: get_is_geom_irred(f,C,[13,269])
+        (False, False)
+
+    This is LMFDB curve 3125.a.3125.1
+
+        sage: f = 4*x^5 + 1
+        sage: C = HyperellipticCurve(f)
+        sage: get_is_geom_irred(f,C,[5])
+        (True, False)
+
+    This is LMFDB curve 277.a.277.2
+
+        sage: f = 4*x^6 - 36*x^4 + 56*x^3 - 76*x^2 + 44*x - 23
+        sage: C = HyperellipticCurve(f)
+        sage: get_is_geom_irred(f,C,[277])
+        (True, True)
+
+    """
+
+    if C.has_odd_degree_model():
+        C_odd = C.odd_degree_model()
+        f_odd, h_odd = C_odd.hyperelliptic_polynomials()
+        # if f was odd to begin with, then f_odd = f
+        assert f_odd.degree() == 5
+
+        if (4*f_odd + h_odd**2).degree() == 5:
+            f_new = 4*f_odd + h_odd**2
+            if f_new.is_irreducible():
+                # i.e. the Jacobian is absolutely simple
+                f_disc_odd_prime_exponents = [v for _,v in f_new.discriminant().prime_to_S_part([ZZ(2)]).factor()]
+                if 1 in f_disc_odd_prime_exponents:
+                    return (True, True)  # Theorem 4.8 (2)
+                return (True, False)  # Algorithm 4.10 Step 1
+
+    if f.is_irreducible():
+        assert f.degree() == 6  # else we should already have exited by now
+        G = f.galois_group()
+        if G.order() in [360, 720]:
+            return (True, True)  # Algorithm 4.10 Step 2
+
+    R = PolynomialRing(ZZ,2,"xv")
+    x,v = R.gens()
+    T = PolynomialRing(QQ,'v')
+    g = v - x**12
+
+    for p in prime_range(3,B):
+        if p not in bad_primes:
+            fp = C.change_ring(FiniteField(p)).frobenius_polynomial()
+
+            # This defines the polynomial f_v**[12] from the paper
+            fp12 = T(R(fp).resultant(g))
+
+            if fp12.is_irreducible():
+                # i.e. the Jacobian is absolutely simpl
+                f_disc_odd_prime_exponents = [v for _,v in f.discriminant().prime_to_S_part([ZZ(2)]).factor()]
+                if 1 in f_disc_odd_prime_exponents :
+                    return (True, True)  # Theorem 4.8 (2)
+                return (True, False)  # Algorithm 4.10 Step 3
+    return (False, False)
+
+
+def is_geom_trivial_when_field(C, bad_primes, B=200):
+    """
```

Comment:
even though this function is not exposed, I would add a one-sentence description of the function
https://doc.sagemath.org/html/en/developer/coding_basics.html#documentation-strings

Reply:
Done.